src/main/java/com/limechain/runtime/RuntimeImpl.java

```diff
@@ -57,6 +62,29 @@ public BabeApiConfiguration getBabeApiConfiguration() {
         return ScaleUtils.Decode.decode(call(RuntimeEndpoint.BABE_API_CONFIGURATION), new BabeApiConfigurationReader());
     }
 
+    @Override
+    public Optional<OpaqueKeyOwnershipProof> generateKeyOwnershipProof(BigInteger slotNumber,
+                                                                       byte[] authorityPublicKey) {
+        byte[] encodedProof = ArrayUtils.addAll(ScaleUtils.Encode.encode(
+                new UInt64Writer(), slotNumber), authorityPublicKey);
+        byte[] encodedResponse = call(RuntimeEndpoint.BABE_API_GENERATE_KEY_OWNERSHIP_PROOF, encodedProof);
+        return new ScaleCodecReader(encodedResponse).readOptional(new OpaqueKeyOwnershipProofReader());
+    }
+
+    @Override
+    public void submitReportEquivocationUnsignedExtrinsic(BlockEquivocationProof blockEquivocationProof,
+                                                          byte[] keyOwnershipProof) {
+        // Key ownership needs to be encoded as a list (have its length encoded) as it is of varying size.
+        try (ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+             ScaleCodecWriter scaleCodecWriter = new ScaleCodecWriter(buffer)) {
```

Comment:
I have a feeling it would be a good idea to have some kind of an encoding util method that can concatenate encoded bytes to one another. In this way we can have chains of writers for a single result. This is just and idea nothing to change here.